src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -0,0 +1,56 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.stereotype.Service;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repositories.FacultyRepository;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+@Service
+public class FacultyService {
+    private final FacultyRepository facultyRepository;
+
+    public FacultyService(FacultyRepository facultyRepository) {
+        this.facultyRepository = facultyRepository;
+    }
+
+    public Faculty createFaculty(Faculty faculty) {
+        return facultyRepository.save(faculty);
+    }
+
+    public Faculty readFaculty(Long id) {
+        if (facultyRepository.existsById(id)) return facultyRepository.findById(id).get();
+        return null;
+    }
+
+    public Faculty updateFaculty(Faculty faculty) {
+        return facultyRepository.save(faculty);
+    }
+
+    public Faculty deleteFaculty(Long id) {
+        Faculty faculty = readFaculty(id);
+        if (faculty == null) return null;
+        facultyRepository.deleteById(id);
+        return faculty;
+    }
+
+    public Collection<Faculty> filterByColor(String color) {
+        return facultyRepository.findAll().stream()
+                .filter(e -> e.getColor().equals(color))
+                .collect(Collectors.toList());
+    }
+
+    public Collection<Faculty> printAll() {
+        return facultyRepository.findAll();
+    }
+
+    public Collection<Faculty> findByNameIgnoreCaseOrColorIgnoreCase(String name, String color) {
+        return facultyRepository.findByNameIgnoreCaseOrColorIgnoreCase(name, color);
+    }
+
+    public Collection<Student> getStudents(Long Id) {
+        return facultyRepository.findById(Id).get().getStudents();
```

Comment:
здесь неверно - будет исключение, если не будет записи в БД по id. НИКОГДА не надо вызывать get у Optional без проверки. Здесь сделать проверку можно одной строкой, если правильно использовать Optional:
```suggestion
        return facultyRepository.findById(Id)
            .map(Faculty::getStudents)
            .orElseGet(Collections::emptyList);
```